contribs/gmf/src/services/permalink.js

```diff
@@ -876,6 +928,58 @@ gmf.Permalink.prototype.updateLayerStateByVisibility_ = function(layer) {
 
 
 /**
+ * Update all properties found in the appplication state for the given layer
+ * @param  {ol.layer.Base} layer The layer
+ * @private
+ */
+gmf.Permalink.prototype.updateLayerFromState_ = function(layer) {
+  var layerName = /** @type {string} */ (layer.get('layerName'));
+  var isMerged = /** @type {boolean} */ (layer.get('isMerged'));
+  var param, stateValue, enable;
+  Object.keys(gmf.PermalinkOpenLayersLayerProperties).forEach(function(layerProp) {
+    param = this.getLayerStateParam_(layerName, isMerged, gmf.PermalinkOpenLayersLayerProperties[layerProp]);
+    stateValue = this.ngeoStateManager_.getInitialValue(param);
+    if (stateValue !== undefined) {
+      layer.set(gmf.PermalinkOpenLayersLayerProperties[layerProp], stateValue);
+    }
+  },this);
+
+  // -- Layer Visibility -- //
+  param = this.getLayerStateParam_(layerName, isMerged);
+  stateValue = this.ngeoStateManager_.getInitialValue(param);
+  if (stateValue !== undefined) {
+    //Visibility state of the layer is not the default one -> user interacted with
+    if (isMerged) {
+      //Not mixed case, we fetched layers names, if not empty: layer must be visible
+      enable = stateValue.length > 0;
+      layer.setVisible(enable);
+    } else {
+      //Mixed case, we fetched true or false
+      enable = stateValue;
+      layer.setVisible(enable);
```

Comment:
yes I know... I can check again but I wrote this because ngeoStateManager_.getInitialValue method can return different types (boolean, string, Object...) and closure was not happy to find it (stateValue) directly in setVisible method which accepts boolean only. Maybe you already encountered this case?